### PR TITLE
feat:order and cart APIs

### PR DESCRIPTION
**Pull request by neha067**
1. Added models,controllers and api definitions for order and cart

### CODE REVIEW

File: server/app/controllers/cart.js [hunk 1]
```diff
@@ -0,0 +1,252 @@
+const Cart = require('../models/cart');
+const { successResponse, errorResponse } = require('../core/response');
+const logger = require('../core/logger');
+
+const { skuGenerator, validateMongoDbId } = require('../utils/utils');
```
Comment (subhamsahu) [inline server/app/controllers/cart.js:5]: skuGenerator is required here ?

General Comment: Remove Unnecessary, Unused Imports

File: server/app/routes/api/order.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+// Bring in Models & Utils
+const auth = require('../../middleware/auth');
+const role = require('../../middleware/role');
+const { checkValidMongoDbId } = require('../../middleware/db');
+const { ROLES } = require('../../constants');
+const orderController = require('../../controllers/order');
+
+// create order
+router.post('/new', auth(), (req, res) => {
```
Comment (subhamsahu) [inline server/app/routes/api/order.js:12]: Can we change route to /create instead of /new

File: server/app/routes/api/order.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  orderController.create(req, res);
+});
+
+// fetch order by order id 
+router.get('/:orderId', auth(),  async (req, res) => {
```
Comment (subhamsahu) [inline server/app/routes/api/order.js:17]: For all mongodb id , put a check for validMongoDbId

File: server/app/models/cart.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const Mongoose = require('mongoose');
+const { Schema } = Mongoose;
+
+// Cart Schema
+const CartSchema = new Schema({
+    userId: {type: Schema.Types.ObjectId, required: true},
+    cartItems: [{
```
Comment (subhamsahu) [inline server/app/models/cart.js:7]: Query Here: are we handling both restaurant and product items here

Comment (neha067) [reply server/app/models/cart.js:7]: Handling both the items together by giving a field 'productType' in 'cartItems' which will have values either 'grocery' or 'restaurant'.

**Review by subhamsahu — APPROVED** [review]

**Pull request merged**